internal/atomicutil/value.go

```diff
@@ -17,7 +17,16 @@ func NewValue[T any](init T) *Value[T] {
 
 // Load loads the value atomically.
 func (v *Value[T]) Load() T {
-	return v.value.Load().(T)
+	var def T
+	if v == nil {
+		return def
+	}
+
+	cur := v.value.Load()
+	if cur == nil {
+		return def
+	}
+	return cur.(T)
```

Comment:
The extra code here was to make it a little more defensive about misuse. The type should always be created with `atomicutil.NewValue` but if its not at least with this added code `.Load` won't panic.